### PR TITLE
Genome filter fix

### PR DESCRIPTION
**Pull request by borismattijssen**
Fixed the genome filter to also filter out the genomes in the nodes that are left over.

### CODE REVIEW

File: src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java [hunk 1]
```diff
@@ -8,12 +8,14 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
-import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.IOException;
```
Comment (TomBrouws) [inline src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java]: pls fix

**Comment by TomBrouws** [comment]
Looks good!

File: src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java (hunk 1 above)
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/graph/GraphDataRepository.java]: maybe

File: src/main/java/tudelft/ti2806/pl3/data/graph/DataNode.java [hunk 2]
```diff
@@ -14,7 +16,7 @@
  */
 public class DataNode {
 	protected final int nodeId;
-	protected final Genome[] source;
+	protected final Set<Genome> source;
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/data/graph/DataNode.java]: Requires a second set for the original data. We are not cloning DataNodes or re-parsing it for each filter.

**Comment by Delfil** [comment]
Duplicate Genome Exception in SpaceWrapUtil is caused by empty edges.
Here an example:
Nodes:
1 A, B
2 A
3 A, B
Edges:
1 2
2 3
1 3
After filtering B out:
Nodes:
1 A
2 A
3 A
Edges:
1 2
2 3
1 3
Now 1 3 is an empty edge, which creates a second possibility for the content of the A genome, which leads to a duplicate genome exception in the SpaceWrapUtil.

**Pull request closed without merging**